kernel-util/src/main/java/org/sakaiproject/content/util/ZipContentUtil.java

```diff
@@ -394,8 +408,26 @@ private void storeContentCollection(String rootId, ContentCollection collection,
 	 * @param out
 	 * @throws Exception
 	 */
-	private void storeContentResource(String rootId, ContentResource resource, ZipOutputStream out) throws Exception {		
-		String filename = resource.getId().substring(rootId.length(),resource.getId().length());				
+	private void storeContentResource(String rootId, ContentResource resource, ZipOutputStream out) throws Exception {
+		String filename = resource.getId().substring(rootId.length(),resource.getId().length());
+		//Inorder to have username as the folder name rather than having eids
+		if(rootId.indexOf("group-user")!=-1 && ServerConfigurationService.getBoolean("dropbox.zip.haveDisplayname", true)) {
+				try {
+					filename = getContainingFolderDisplayName(rootId, filename);
+				} catch(TypeException e){
+					LOG.warn("Unexpected error occurred when trying to create Zip archive:" + extractName(rootId)+e.getStackTrace());
```

Comment:
If it's something the developer needs to debug, you're much better off passing the stack trace as a second argument to the log call, ie `LOG.warn(String,Throwable)`.